python/looker_sdk/rtl/transport.py

```diff
@@ -45,8 +46,7 @@ class HttpMethod(enum.Enum):
     HEAD = 7
 
 
-@attr.s(auto_attribs=True, kw_only=True)
-class TransportSettings:
+class PTransportSettings(Protocol):
```

Comment:
oh goodness, so I started a deep dive into https://www.python.org/dev/peps/pep-0544/#protocol-members and the next section https://www.python.org/dev/peps/pep-0544/#explicitly-declaring-implementation ...

my basic understanding is that you can actually use protocol classes as baseclasses (**explicit** subtyping) and get the method implementations (e.g. `is_configured`) and default properties for free. However, because we're throwing `@attr.s` classes into the mix I feel like we should stick with **implicit** subtyping in which case we get rid of the member defaults and leave the `is_configured` method body as `...`

I think we can also keep a concrete settings class here too. so something like:

```python
class PTransportSettings(Protocol):
    base_url: str
    api_version: str
    verify_ssl: bool
    timeout: int
    headers: Optional[MutableMapping[str, str]]

    def is_configured(self) -> bool:
        ...


@attr.s(auto_attribs=True, kw_only=True)
class TransportSettings:
    base_url: str = ""
    api_version: str = "3.1"
    verify_ssl: bool = True
    timeout: int = 120
    headers: Optional[MutableMapping[str, str]] = None

    def is_configured(self) -> bool:
        return bool(self.base_url and self.api_version)
```

and then in api_settings.py leave `PApiSettings` exactly how you have it (add `section: Optional[str]` property) and change `ApiSettings` to

```python
@attr.s(auto_attribs=True, kw_only=True)
class ApiSettings(transport.TransportSettings:
    section: Optional[str] = None

    @classmethod
    def configure(...)
    .
    .
    .
     # keep all the remaining code but DITCH is_configured()
```

Let me know if that's not entirely clear. Basically we want to reflect the fact that the transport module can be standalone with it's own settings interface/type and the api_settings module has some extra stuff